litex/soc/software/bios/commands/cmd_boot.c

```diff
@@ -0,0 +1,107 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <generated/csr.h>
+
+#include "../command.h"
+#include "../helpers.h"
+#include "../boot.h"
+
+/**************************************
+ *            Flash boot              *
+ **************************************
+ */
+#ifdef FLASH_BOOT_ADDRESS
+static void flashboot_handler(int nb_params, char **params)
+{
+	flashboot();
+}
+
+struct command_struct cmd_flashboot =
+{
+	.func = flashboot_handler,
+	.name = "flashboot",
+	.help = "Boot from flash",
+};
+
+define_command(cmd_flashboot);
+#endif
+
+/**************************************
+ *              ROM boot              *
+ **************************************
+ */
+#ifdef ROM_BOOT_ADDRESS
+static void romboot_handler(int nb_params, char **params)
+{
+	romboot();
+}
+
+struct command_struct cmd_romboot =
+{
+	.func = romboot_handler,
+	.name = "romboot",
+	.help = "Boot from embedded rom",
+};
+
+define_command(cmd_romboot);
+#endif
+
+/**************************************
+ *            Serial boot             *
+ **************************************
+ */
+static void serialboot_handler(int nb_params, char **params)
+{
+	serialboot();
+}
+
+struct command_struct cmd_serialboot =
+{
+	.func = serialboot_handler,
+	.name = "serialboot",
+	.help = "Boot via SFL",
+};
+
+define_command(cmd_serialboot);
+
+/**************************************
+ *          Ethernet boot             *
+ **************************************
```

Comment:
I do like that this kind of structure also leads the way towards `liteeth` and `litedram` providing the BIOS features as compile time plugins rather than it all being in stuck in the litex repo.